spv/backend.go

```diff
@@ -89,6 +91,48 @@ func (s *Syncer) CFiltersV2(ctx context.Context, blockHashes []*chainhash.Hash)
 	}
 }
 
+// cfiltersV2FromNodes fetches cfilters for all the specified nodes from a
+// remote peer.
+func (s *Syncer) cfiltersV2FromNodes(ctx context.Context, cnet wire.CurrencyNet, rp *p2p.RemotePeer, nodes []*wallet.BlockNode) ([]*gcs.FilterV2, error) {
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
maybe add a check for len(nodes) == 0 here.  The net effect is the same, but the calling code is not doing this check prior to calling this function, and we can avoid creating the derived context if no cflters are needed.